components/Router.php

```diff
@@ -48,24 +44,37 @@ public function run():void {
 						 * 
 						 * @var string $id
 						 */
-						$id = $matches[1] ?? '';
+						$id         = $matches[1] ?? "";
+						$invokeData = compact("factory", "action", "id");
 
-						/**
-						 * Factory used for creating MVCPDM components
-						 * 
-						 * @var IMVCPDMFactory
-						 */
-						$factoryObj = new $factory();
-
-						$proxyController = $factoryObj->getProxy();
-
-						if ($id) {
-							$proxyController->$action($id);
-						} else {
-							$proxyController->$action();
-						}
+						$this->invokeFactory($invokeData);
 					}
 				}
 			}
-        }
+		}
+		
+		/**
+		 * Creates factory and invokes its Proxy Controller method
+		 *
+		 * @param string[] $invokeData
+		 * @return void
+		 */
+		protected function invokeFactory(array $invokeData = []):void {
+			extract($invokeData);
+
+			/**
+			 * Factory used for creating MVCPDM components
+			 * 
+			 * @var IMVCPDMFactory
+			 */
+			$factoryObj = new $factory();
+
+			$proxyController = $factoryObj->getProxy();
+
+			if ($id) {
+				$proxyController->$action($id);
+			} else {
+				$proxyController->$action();
+			}
+		}
```

Comment:
The initial 'run' method was too long so the factory call was extracted into a separate method